broker/src/main/java/io/zeebe/broker/system/SystemContext.java

```diff
@@ -119,6 +122,10 @@ private void validateConfiguration() {
               "diskUsageCommandWatermark (%f) must be less than diskUsageReplicationWatermark (%f)",
               diskUsageCommandWatermark, diskUsageReplicationWatermark));
     }
+
+    if (replicationFactor > 1 && experimental.isDisableExplicitRaftFlush()) {
+      LOG.warn(REPLICATION_WITH_DISABLED_FLUSH_WARNING);
```

Comment:
It is not safe to enable this without replication also. I would warn also for replicationFactor = 1.

Reply:
@deepthidevaki how come?

Reply:
@korthout When a write is not flushed to the disk, it is not guaranteed to be persisted over a failure. So it can happen that we write a command, raft commits it but not flush it, processor process it and send a response to the client, and then the node is crashed. After restart, that command does not exists in the log because it was not flushed. So we might have responded to the client that a workflow was created, but after a failure that workflow does not exists.

Reply:
@deepthidevaki thanks. That makes a lot of sense